### PR TITLE
Update Istio Configuration Docs

### PR DESCRIPTION
**Pull request by Tomasz-Smelcerz-SAP**
**Description**

Update documentation with information about new Istio overrides.

Changes proposed in this pull request:

- Update documentation about new Istio overrides.

**Related issue(s)**
See also #7129

### CODE REVIEW

File: docs/service-mesh/03-01-istio-setup-in-kyma.md [hunk 1]
```diff
@@ -3,7 +3,10 @@ title: Istio setup in Kyma
 type: Details
 ---
 
-Istio is installed using the official charts from the currently supported release. The charts are customized for Kyma and are stored in the `resources/istio` directory.
+Istio in Kyma is installed with the help of the `istioctl` tool.
+The tool is driven by a configuration file containing an instance of [IstioControlPlane](https://istio.io/docs/reference/config/istio.operator.v1alpha12.pb) custom resource.
```
Comment (majakurcius) [inline docs/service-mesh/03-01-istio-setup-in-kyma.md]:
```suggestion
The tool is driven by a configuration file containing an instance of the [IstioControlPlane](https://istio.io/docs/reference/config/istio.operator.v1alpha12.pb) custom resource.
```

File: docs/service-mesh/03-01-istio-setup-in-kyma.md [hunk 2]
```diff
@@ -3,7 +3,10 @@ title: Istio setup in Kyma
...
+There are two configuration files, one for local installation on Minikube and one for cluster installations.
```
Comment (majakurcius) [inline docs/service-mesh/03-01-istio-setup-in-kyma.md]:
```suggestion
There are two configuration files — one for local installation on Minikube and one for cluster installations.
```

File: docs/service-mesh/05-05-production-profile.md [hunk 3]
```diff
@@ -22,18 +22,108 @@ The production profile introduces the following changes to the Istio Service Mes
...
+|:--- | :--- | :--- |
+| vCPU | 4 | 8 |
+| RAM | 16 | 16/32 |
+| Example machine type (GKE) | `n1-standard-4` | `n1-standard-8` or `c2-standard-8` |
+| Example machine type (AKS) | `Standard_D4_v3` | `Standard_F8s_v2` or `Standard_D8_v3` |
+| Size | 3 | 3-5 |
 
 ## Use the production profile
 
+>**CAUTION:** Istio is changing the installation options. Helm-based configuration is deprecated in favor of the new IstioControlPlane API. Please keep in mind that Helm overrides will be no longer supported in future Istio releases. Refer to [IstioControlPlane](https://istio.io/docs/reference/config/istio.operator.v1alpha12.pb) page for more information.
```
Comment (majakurcius) [inline docs/service-mesh/05-05-production-profile.md]:
```suggestion
>**CAUTION:** Istio is changing the installation options. Helm-based configuration is deprecated in favor of the new IstioControlPlane API. Please keep in mind that Helm overrides will no longer be supported in future Istio releases. Refer to [IstioControlPlane](https://istio.io/docs/reference/config/istio.operator.v1alpha12.pb) page for more information.
```
Also, I wonder about this sentence:
> Istio is changing the installation options. 

It creates a risk of the document getting outdated once Istio's done with changing the installation options if you fail to remember to update it. 
Maybe rephrase it to something like `Due to changes in the installation options in Istio, Helm-based configuration is now deprecated in favor of the new IstioControlPlane API.`?

File: docs/service-mesh/05-05-production-profile.md [hunk 4]
```diff
@@ -22,18 +22,108 @@ The production profile introduces the following changes to the Istio Service Mes
...
 You can deploy a Kyma cluster with Istio configured to use the production profile, or configure Istio in a running cluster to use the production profile. Follow these steps:
 
 <div tabs>
+  <details>
+  <summary>
+  Istio Control Plane API
+  </summary>
+Istio installation in Kyma uses the [IstioControlPlane](https://istio.io/docs/reference/config/istio.operator.v1alpha12.pb) API.
+This API is in the alpha version, but it's going to replace Helm-based approach in future Istio versions.
+Kyma provides a default IstioControlPlane configurations for local (Minikube) and cluster installations.
```
Comment (majakurcius) [inline docs/service-mesh/05-05-production-profile.md]:
```suggestion
Kyma provides the default IstioControlPlane configurations for local (Minikube) and cluster installations.
```

File: docs/service-mesh/05-05-production-profile.md [hunk 5]
```diff
@@ -22,18 +22,108 @@ The production profile introduces the following changes to the Istio Service Mes
...
+You can add a custom control plane definition that overrides the default settings.
+The definition you provide may be a partial one (you don't have to specify all options), it will be merged with the defaults.
```
Comment (majakurcius) [inline docs/service-mesh/05-05-production-profile.md]:
```suggestion
The definition you provide may be a partial one (you don't have to specify all options). In that case it will be merged with the defaults.
```

File: docs/service-mesh/05-05-production-profile.md [hunk 6]
```diff
@@ -22,18 +22,108 @@ The production profile introduces the following changes to the Istio Service Mes
...
+In order to provide a custom IstioControlPlane configuration, define a Kyma Installation override with a key: `kyma_istio_control_plane`.
+The value for this override must be a single string containing a valid definition of IstioControlPlane custom resource, in a YAML format.
```
Comment (majakurcius) [inline docs/service-mesh/05-05-production-profile.md]:
```suggestion
The value for this override must be a single string containing a valid definition of the IstioControlPlane custom resource, in the YAML format.
```

File: docs/service-mesh/05-05-production-profile.md [hunk 7]
```diff
@@ -22,18 +22,108 @@ The production profile introduces the following changes to the Istio Service Mes
...
+  <details>
+  <summary>
+  Istio Control Plane API
+  </summary>
+Istio installation in Kyma uses the [IstioControlPlane](https://istio.io/docs/reference/config/istio.operator.v1alpha12.pb) API.
+This API is in the alpha version, but it's going to replace Helm-based approach in future Istio versions.
+Kyma provides a default IstioControlPlane configurations for local (Minikube) and cluster installations.
+You can add a custom control plane definition that overrides the default settings.
+The definition you provide may be a partial one (you don't have to specify all options), it will be merged with the defaults.
+In order to provide a custom IstioControlPlane configuration, define a Kyma Installation override with a key: `kyma_istio_control_plane`.
```
Comment (majakurcius) [inline docs/service-mesh/05-05-production-profile.md]:
```suggestion
In order to provide a custom IstioControlPlane configuration, define a Kyma Installation override with the `kyma_istio_control_plane` key.
```

File: docs/service-mesh/05-05-production-profile.md [hunk 8]
```diff
@@ -22,18 +22,108 @@ The production profile introduces the following changes to the Istio Service Mes
...
+The value for this override must be a single string containing a valid definition of IstioControlPlane custom resource, in a YAML format.
+
+>**TIP:** To learn more about how to use overrides in Kyma, see the following documents:
+>* [Helm overrides for Kyma installation](/root/kyma/#configuration-helm-overrides-for-kyma-installation)
+>* [Top-level charts overrides](/root/kyma/#configuration-helm-overrides-for-kyma-installation-top-level-charts-overrides)
```
Comment (majakurcius) [inline docs/service-mesh/05-05-production-profile.md]:
```suggestion
>* [Top-level charts overrides](/root/kyma/#configuration-helm-overrides-for-kyma-installation-top-level-charts-overrides).
```

File: docs/service-mesh/05-05-production-profile.md (hunk 3 above)
Comment (Tomasz-Smelcerz-SAP) [reply docs/service-mesh/05-05-production-profile.md]: Suggestion applied, thanks.
